elide-core/src/test/java/com/yahoo/elide/core/filter/dialect/RSQLFilterDialectTest.java

```diff
@@ -512,4 +517,35 @@ public void testMemberOfOperatorOnRelationshipException() throws Exception {
 
         assertEquals("Invalid Path: Last Path Element cannot be a collection type", e.getMessage());
     }
+
+    @Test
+    public void testFilterArgumentParsing() throws Exception {
+
+        FilterExpression expr;
+        Type<Book> bookType = new ClassType<Book>(Book.class);
+
+        expr = assertDoesNotThrow(() -> dialect.parse(bookType, Collections.emptySet(),
+                        "genre=in=(sci-fi,action),title[a:b][x:y]==Hemingway", NO_VERSION));
+        assertEquals("(book.genre IN [sci-fi, action] OR book.title IN [Hemingway])", expr.toString());
+
+        // Empty argument name or value
+        expr = assertDoesNotThrow(() -> dialect.parse(bookType, Collections.emptySet(),
+                        "genre=in=(sci-fi,action),title[a:][:y]==Hemingway", NO_VERSION));
+        assertEquals("(book.genre IN [sci-fi, action] OR book.title IN [Hemingway])", expr.toString());
+
+        // Missing argument name or value (: missing within [])
+        Exception e = assertThrows(ParseException.class, () -> dialect.parse(bookType, Collections.emptySet(),
+                        "genre=in=(sci-fi,action),title[ab][x:y]==Hemingway", NO_VERSION));
+        assertEquals("Failed to parse arguments from filter expression at: title[ab][x:y]", e.getMessage());
+
+        // Anything after first '[' and not within [] is ignored
```

Comment:
I think we should be strict here and throw an exception (see earlier comment on refining the regex).